packages/neo-one-client-core/src/user/keystore/LedgerHandler.ts

```diff
@@ -0,0 +1,146 @@
+import { crypto, PublicKeyString } from '@neo-one/client-common';
+import { HWLedger, LedgerTransport } from '@neo-one/client-switch';
+import { LedgerNotDetectedError, LedgerNotSupportedError } from '../../errors';
+
+export interface LedgerResponse {
+  readonly message: Buffer;
+  readonly status: Buffer;
+}
+
+export interface LedgerQueueRequest {
+  readonly msg: Buffer;
+  readonly resolve: (value: LedgerResponse) => void;
+  readonly reject: (error: Error) => void;
+}
+
+export enum Statuses {
+  VALID_STATUS = 0x9000,
+  // MSG_TOO_BIG = 0x6d08,
+  // APP_CLOSED = 0x6e00,
+  // TX_DENIED = 0x6985,
+  // TX_PARSE_ERR = 0x6d07,
+}
+
+const keyRequestParams = Buffer.from('80040000', 'hex');
+const initMessageParams = Buffer.from('80020000', 'hex');
+const finalMessageParams = Buffer.from('80028000', 'hex');
+
+const BIP44 = (accountIn = 0): Buffer => {
+  const account = accountIn.toString(16);
+
+  return Buffer.from(`8000002C800003788000000000000000${'0'.repeat(8 - account.length)}${account}`, 'hex');
+};
+
+const chunkBuffer = (buffer: Buffer, chunkLength: number): ReadonlyArray<Buffer> => {
+  if (buffer.length > chunkLength) {
+    return [buffer.slice(0, chunkLength)].concat(chunkBuffer(buffer.slice(chunkLength), chunkLength));
+  }
+
+  return [buffer];
+};
+
+export class LedgerHandler {
+  public static readonly byteLimit = LedgerTransport.byteLimit;
+  public static readonly type = LedgerTransport.type;
+
+  public static readonly init = async () => {
+    const isSupported = await LedgerTransport.isSupported();
+    if (!isSupported) {
+      throw new LedgerNotSupportedError();
+    }
+    const paths = await LedgerTransport.list();
+    if (paths.length === 0) {
+      throw new LedgerNotDetectedError();
+    }
+    const ledger = await LedgerTransport.open(paths[0]);
+    ledger.setScrambleKey('NEO');
+
+    return new LedgerHandler(ledger);
+  };
+
+  private readonly ledger: HWLedger;
+  private readonly mutableQueue: LedgerQueueRequest[];
+  private mutableRunning: boolean;
+
+  private constructor(ledger: HWLedger) {
+    this.ledger = ledger;
+    this.mutableRunning = false;
+    this.mutableQueue = [];
+  }
+
+  public async getPublicKey(account = 0): Promise<PublicKeyString> {
+    const response = await this.send(Buffer.concat([keyRequestParams, Buffer.from('14', 'hex'), BIP44(account)]));
```

Comment:
This should also sign the public key and then verify that the message (i.e. the publickey) is correctly signed. This is what the neon folks are doing since there seems to be a small chance that the ledger returns the incorrect public key.

Reply:
Could you elaborate? Their [getPublicKey](https://github.com/CityOfZion/neon-wallet/blob/78a43dc05182d0504c0c16107ae92bde2287e21e/app/ledger/neonLedger.js#L100) code seems to be the same as ours.

Reply:
oh, never mind, guess they aren't doing that anymore.